[hpcman] iP

hpcman opened the pull request:
# PacManPro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

PacManPro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is __FREE__!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the <code>main</code> method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Rename main class, add greeting and exiting

Commit:
Add echoing function

Commit:
Add listing function, change function access modifier

Commit:
Add marking task function, use echoing function as main output function

Commit:
Add todo, event, deadline feature; fix minor bug

Commit:
Add automated text UI testing

Commit:
Add errors handlers for mark and unmark

Commit:
Add delete task feature

Commit:
enchantment: Update conversation and change switch usage to if-else for clarity

Commit:
bug fix: Add event insertion function

Commit:
Update EXPECTED.TXT

WuJinhan1 commented on src/main/java/PacMan.java:
```diff
@@ -0,0 +1,120 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class PacMan {
```
I like how you named your chatbot PacMan! Putting a figure behind the chatbot gives it more character.

WuJinhan1 commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,25 @@
-
+  Hello! I'm PacMan. How can I help you?
+  Got it. I've added this task:
+    [T][ ] CS2103T iP
```
I like how you made the tasks relevant to the class we are taking.

WuJinhan1 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,18 @@
+public class Task {
+    private final String taskName;
```
LGTM!

WuJinhan1 commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,10 @@
+public class Todo extends Task {
+    Todo(String task) {
```
I like how you split up each type of task into different files, making the code really clean!

WuJinhan1 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
```
Should all of this be in a package?

WuJinhan1 reviewed (commented):
Overall, LGTM! Just suggested that the files should be put inside a package.

Commit:
add feature: save task list in the filesystem

Commit:
Revert "add feature: save task list in the filesystem"

This reverts commit 39d5276a8fe8e3be842e5e389399d12a1f682b36.

Commit:
add feature: save task list in the filesystem

Commit:
Deadline class: Change time format to LocalDate

Commit:
Event class: Change time format to LocalDate

Commit:
PacMan class: Cut from list for easier LocalDate parse and change variable name from to by in addDeadline method for readability

Commit:
Edit input.txt and EXPECTED.txt to follow the time format change

Commit:
Add Ui.java

Commit:
Add TaskList.java

Commit:
Add Storage.java

Commit:
Add Parser.java

Commit:
PacMan class: Update main method to use additional class

Commit:
Move Deadline.java, Event.java, Parser.java, Storage.java, Task.java, TaskList.java, Todo.java, Ui.java to PacMan folder and bundle them to packages

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
build.gradle: Update default class to PacMan

Commit:
Move PacMan.java

Commit:
Update build.gradle

Commit:
Rename package name to follow format

Commit:
Add StorageTest.java

Commit:
Add TaskListTest.java

Commit:
build.gradle: Modify shadowJar

Commit:
Deadline.java: Add JavaDoc

Commit:
Event.java: Add JavaDoc

Commit:
Parser.java: Add JavaDoc

Commit:
Storage.java: Add JavaDoc

Commit:
Task.java: Add JavaDoc

Commit:
TaskList.java: Add JavaDoc

Commit:
Todo.java: Add JavaDoc

Commit:
Ui.java: Add JavaDoc

Commit:
Deadline.java: Fix style issues

Commit:
Event.java: Fix style issues

Commit:
Storage.java: Fix style issues

Commit:
Todo.java: Fix style issues

Commit:
Add find feature

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/PacMan/Deadline.java
#	src/main/java/PacMan/Event.java
#	src/main/java/PacMan/Storage.java
#	src/main/java/PacMan/Todo.java

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/PacMan/Parser.java
#	src/main/java/PacMan/TaskList.java

izruff reviewed (commented):
Overall, the code base is fully functional but needs some improvement, especially on error handling. My suggestion is to define exception classes and modify your code so there is a better separation of concerns.

izruff commented on src/main/java/PacMan/Parser.java:
```diff
@@ -0,0 +1,128 @@
+            try {
+                markTask(Integer.parseInt(command.split(" ")[1]));
+            } catch (ArrayIndexOutOfBoundsException e) {
+                ui.showResult("I'm sorry, but I can't find the index number :(");
```
I noticed you do not have classes related to error handling. One way to improve your code structure is to separate error handling by defining your own exception classes. In this case, you could throw a new error such as `TaskIndexOutOfBoundsException`, and let the `Ui` instance handle it instead of immediately printing the error.

izruff commented on src/main/java/PacMan/Todo.java:
```diff
@@ -0,0 +1,25 @@
+     *
+     * @return a <code>String</code> that is readable and writeable by <code>Storage</code>
+     */
+    public String toFile() {
```
I like how you used the `Task` superclass here. Good application of inheritance and code reusability.

Commit:
Merge branch 'branch-Level-7'

Commit:
Delete package PacMan

Commit:
Add package pacman

Commit:
Delete PacMan.java

Commit:
Add Pacman.java

Commit:
Add Ghost.png and Pacman.png

Commit:
Pacman.java: Rename class PacMan to class Pacman

Commit:
build.gradle: Add JavaFX dependencies and update start point

Commit:
Add GUI

Commit:
Pacman.java: Migrate to GUI

Commit:
Ui.java: Migrate to GUI

Commit:
Add checkStyle support

Commit:
bug fix: Change pacman.png and ghost.png to Pacman.png and Ghost.png

Commit:
bug fix: Use default exception to handle invalid user input

Commit:
Storage.java: Add assertion on invalid data

Commit:
Task.java: Change markDone to isMarkDone

Commit:
Merge branch 'branch-A-CodeQuality'

hpcman committed:
Better GUI

The past GUI is bare bone and not visually appealing.

The GUI needs to be better for visual appeal and easier to separate the command
and response.

Let's,
* use CSS FXML to make the background and dialog box

hpcman committed:
bug fix: Add 's' for all completed JavaDoc

hpcman committed:
bug fix: Change Java version from 22 to 17

hpcman committed:
dialog-box.css: Remove unused line

hpcman committed:
Merge pull request #2 from hpcman/branch-A-CodeQuality

Branch a code quality

hpcman committed:
Parser.java: Support simpler syntax

hpcman committed:
Move MainWindow.java to package pacman.gui

hpcman committed:
Add Ui.png

hpcman committed:
README.md: Add user guide

hpcman committed:
Main.java: Migrate MainWindow from pacman to pacman.gui

hpcman committed:
Pacman.java: Fix naming issue

hpcman committed:
DialogBox.java: Change to fit MainWindow package change

hpcman committed:
MainWindow.fxml: Change to fix MainWindow package change

hpcman committed:
bug fix: Add additional shorter command and remove trailing next line

hpcman committed:
Add exception

hpcman committed:
Fix style issues
